go/pools/resource_pool.go

```diff
@@ -104,9 +104,12 @@ func (rp *ResourcePool) TryGet() (resource Resource, err error) {
 
 func (rp *ResourcePool) get(ctx context.Context, wait bool) (resource Resource, err error) {
 	// If ctx has already expired, avoid racing with rp's resource channel.
-	if ctx.Err() != nil {
+	select {
+	case <-ctx.Done():
```

Comment:
@sougou actually authored this fix ;-) But I was too lazy to specify this in the git commit. I'll do that the next time :)

To sum up, the change is fine with @sougou and I'll merge it now.